libraries/standard/mqtt/src/iot_mqtt_validate.c

```diff
@@ -147,52 +169,37 @@ static bool _validateString( const char * pString,
     if( pString == NULL )
     {
         status = false;
-        goto cleanup;
     }
-
-    if( length == 0 )
+    else if( length == 0 )
     {
         status = false;
-        goto cleanup;
     }
-
-cleanup:
+    else
+    {
+        status = true;
+    }
 
     return status;
 }
 
 /*-----------------------------------------------------------*/
 
-static bool _validatePublish( bool awsIotMqttMode,
-                              size_t maximumPayloadLength,
-                              const char * pPublishTypeDescription,
-                              const IotMqttPublishInfo_t * pPublishInfo )
+static bool _validatePublishPayload( const IotMqttPublishInfo_t * pPublishInfo,
+                                     size_t maximumPayloadLength,
+                                     const char * pPublishTypeDescription )
 {
     bool status = true;
 
     /* This parameter is not used when logging is disabled. */
     ( void ) pPublishTypeDescription;
 
-    /* Check for NULL. */
     if( pPublishInfo == NULL )
```

Comment:
Is this necessary? At line 257, `_validatePublish` only calls `_validatePublishPayload` when `pPublishInfo != NULL`

Reply:
That's correct, but this is independent function and may be used later somewhere else (and may get flagged since we are checking for NULL)